astropy/io/fits/diff.py

```diff
@@ -1472,56 +1463,6 @@ def diff_values(a, b, rtol=0.0, atol=0.0):
         return a != b
 
 
-def report_diff_values(fileobj, a, b, ind=0):
-    """Write a diff between two values to the specified file-like object."""
-
-    typea = type(a)
-    typeb = type(b)
-
-    if (isinstance(a, str) and not isinstance(b, str)):
-        a = repr(a).lstrip('u')
-    elif (isinstance(b, str) and not isinstance(a, str)):
-        b = repr(b).lstrip('u')
```

Comment:
BTW, as I was trying to understand what this function does, this part looks weird: the `lstrip('u')` was probably useful only for Python 2, to handle str and unicode strings the same way. 
And it seems to me than the wrong variable is converted to str: if a is str and not b, here it convert a, which makes no sense to me.

Reply:
Why does it make no sense? If the logic is specifically targeting unicode, this seems correct?
```python
>>> a = u'1234'
>>> isinstance(a, str)
True
```
You are probably right about `lstrip('u')` but I am scared to remove it. I don't want to break `FITSDiff`. As a wise person once said, "I should not touch FITS."

Reply:
Ah right, this is just to get the representation of the string with the leading `u`, but this was useful only for Python 2 so the whole block can be removed I guess (if someone brave enough wants to try :wink: )

Reply:
Maybe as a separate PR? @mcara already has a refactoring in mind. :sweat_smile:

Reply:
OK, I can do that too. @pllim asked me to wait until this is merged.

Reply:
Yes, I did. To work on it now concurrently would introduce a lot of conflicts.

Reply:
I do not disagree and I actually understand this. Just saying that I will wait.

Reply:
Yes, let's keep changes for a follow-up PR :+1: